daemon/cluster/swarm.go

```diff
@@ -228,6 +228,13 @@ func (c *Cluster) Update(version uint64, spec types.Spec, flags types.UpdateFlag
 			return err
 		}
 
+		// Validate spec name.
+		if spec.Annotations.Name == "" {
+			spec.Annotations.Name = "default"
+		} else if spec.Annotations.Name != "default" {
+			return errors.New(`swarm spec must be named "default"`)
```

Comment:
At first I thought this might ought to go in `daemon/cluster/convert/swarm.go`, but after stewing on it for a few minutes, I think this is an equally good place. Doesn't mix validation and conversion logic. LGTM.

Reply:
I change my mind lol this actually does need to be changed to wrap this error in one of the errors in `daemon/cluster/errors.go`. If you just return a raw error, it'll probably end up being an 500 Internal at the HTTP api boundary.